portfolio/src/main/java/com/google/sps/data/BlogHashMap.java

```diff
@@ -60,26 +61,26 @@ public void putInMap(List<BlogMessage> messages) {
 
   // Returns a requested amount of all BlogMessages in map
   private LinkedList<BlogMessage> getMessages(int limit) {
-    LinkedList values = new LinkedList();
-
-    for (LinkedList<BlogMessage> tagMessages : map.values()){
+    LinkedList<BlogMessage> values = new LinkedList<BlogMessage>();
+
+    // |valuesList| is a list of LinkedList BlogMessages for each tag in map, hence
+    // We reverse iterate starting from the last tag entered.
+    List valuesList = new ArrayList(map.values());
```

Comment:
Use the complete type List<List<String>> to make the type clear so that you don't need the comments in lines 66-67?

Reply:
My bad, I meant List<LinkedList<String>>

Reply:
Okay I made two mistakes in a row. For map.values(), I think it should be List<LinkedList<BlogMessage>>.